django/core/management/commands/loaddata.py

```diff
@@ -222,57 +243,67 @@ def load_label(self, fixture_label):
                     RuntimeWarning
                 )
 
+    def _get_fixture_name_and_dirs(self, fixture_name):
+        dirname, basename = os.path.split(fixture_name)
+        if os.path.isabs(fixture_name):
+            fixture_dirs = [dirname]
+        else:
+            fixture_dirs = self.fixture_dirs
+            if os.path.sep in os.path.normpath(fixture_name):
+                fixture_dirs = [os.path.join(dir_, dirname) for dir_ in fixture_dirs]
+        return basename, fixture_dirs
+
+    def _get_targets(self, fixture_name, ser_fmt, cmp_fmt):
+        databases = [self.using, None]
+        cmp_fmts = list(self.compression_formats) if cmp_fmt is None else [cmp_fmt]
+        ser_fmts = self.serialization_formats if ser_fmt is None else [ser_fmt]
+        suffixes = (
+            '.'.join(ext for ext in combo if ext)
+            for combo in product(databases, ser_fmts, cmp_fmts)
+        )
+        return {'.'.join((fixture_name, suffix)) for suffix in suffixes}
```

Comment:
As far as I'm aware we don't need to iterate twice over the same list:
```suggestion
        return { 
            '%s.%s' % (
                fixture_name,
                '.'.join([ext for ext in combo if ext]),
            ) for combo in product(databases, ser_fmts, cmp_fmts)
        }
```

Reply:
For some reason GitHub wouldn't let me hit "Commit suggestion", but 455aea843c7f46e8c451d10fb5db6eb6073d1e2a should be identical to your suggestion. As with your other suggestion, I hadn't originally done it to minimize the diff, but happy to now that you've reviewed.